airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/bump_version/pipeline.py

```diff
@@ -31,29 +34,73 @@ def get_bumped_version(version: str, bump_type: str) -> str:
     return str(new_version)
 
 
+class RestoreVersionState(Step):
+    context: ConnectorContext
+
+    title = "Restore original version state"  # type: ignore
+
+    def __init__(self, context: ConnectorContext) -> None:
+        super().__init__(context)
+        connector = context.connector
+        if connector.metadata_file_path.is_file():
+            self.metadata_content = connector.metadata_file_path.read_text()
+        if connector.dockerfile_file_path.is_file():
+            self.dockerfile_content = connector.dockerfile_file_path.read_text()
+        if connector.poetry_file_path.is_file():
+            self.poetry_content = connector.poetry_file_path.read_text()
+        if connector.documentation_file_path and connector.documentation_file_path.is_file():
+            self.documentation_content = connector.documentation_file_path.read_text()
+
+    async def _run(self) -> StepResult:  # type: ignore
+        connector = self.context.connector
+        if self.metadata_content:
+            connector.metadata_file_path.write_text(self.metadata_content)
+        if self.dockerfile_content:
+            connector.dockerfile_file_path.write_text(self.dockerfile_content)
+        if self.poetry_content:
+            connector.poetry_file_path.write_text(self.poetry_content)
+        if self.documentation_content and connector.documentation_file_path:
+            connector.documentation_file_path.write_text(self.documentation_content)
+
+    async def _cleanup(self) -> StepResult:
+        return StepResult(step=self, status=StepStatus.SUCCESS)
+
+        return StepResult(
+            step=self,
+            status=StepStatus.SUCCESS,
+        )
+
+
 class AddChangelogEntry(Step):
     context: ConnectorContext
-    title = "Add changelog entry"
+    title = "Add changelog entry"  # type: ignore
 
     def __init__(
         self,
         context: ConnectorContext,
-        repo_dir: Container,
         new_version: str,
         comment: str,
         pull_request_number: str,
-        export_docs: bool = False,
+        repo_dir: Directory | None = None,
+        export: bool = True,
     ) -> None:
         super().__init__(context)
         self.repo_dir = repo_dir
         self.new_version = semver.VersionInfo.parse(new_version)
         self.comment = comment
         self.pull_request_number = int(pull_request_number)
```

Comment:
I've hit a snag with this when upstream `pull_request_number` is a string, but here we hard-convert it to int, and we will crash if it does not cast. I _think_ all those PR numbers come either from CLI options or generated internally with a command, we should probably take `int` in the action initializer. /cc @alafanechere 

Happy to writeup an issue if you'd like.

Reply:
To be clear, not suggesting @bleonard fixes it in this PR. In fact, definitely don't do that in this PR.